plexapi/server.py

```diff
@@ -607,6 +616,105 @@ def _allowMediaDeletion(self, toggle=False):
+
+            Raises:
+                :exc:`~plexapi.exceptions.BadRequest`: When applying an invalid timespan or unknown filter.
+
+            Example:
+
+                .. code-block:: python
+
+                    from plexapi.server import PlexServer
+                    plex = PlexServer('http://localhost:32400', token='xxxxxxxxxxxxxxxxxxxx')
+
+                    # Filter bandwidth data for December 2020 and later, and more than 1 GB used.
+                    filters = {
+                        'at>': datetime(2020, 12, 1),
+                        'bytes>': 1024**3
+                    }
+
+                    # Retrieve bandwidth data in one day timespans.
+                    bandwidthData = plex.bandwidth(timespan='days', **filters)
+
+                    # Print out bandwidth usage for each account and device combination.
+                    for bandwidth in sorted(bandwidthData, key=lambda x: x.at):
+                        account = bandwidth.account()
+                        device = bandwidth.device()
+                        gigabytes = round(bandwidth.bytes / 1024**3, 3)
+                        local = 'local' if bandwidth.lan else 'remote'
+                        date = bandwidth.at.strftime('%Y-%m-%d')
+                        print('%s used %s GB of %s bandwidth on %s from %s'
+                              % (account.name, gigabytes, local, date, device.name))
+
+        """
+        params = {}
+
+        if timespan is None:
+            params['timespan'] = 6  # Default to seconds
+        else:
+            timespans = {
+                'seconds': 6,
+                'hours': 4,
+                'days': 3,
+                'weeks': 2,
+                'months': 1
+            }
+            try:
+                params['timespan'] = timespans[timespan]
+            except KeyError:
+                raise BadRequest('Invalid timespan specified: %s. '
+                    'Available timespans: %s' % (timespan, ', '.join(timespans.keys())))
+
+        filters = {'accountID', 'at', 'at<', 'at>', 'bytes', 'bytes<', 'bytes>', 'deviceID', 'lan'}
+
+        for key, value in kwargs.items():
+            if key not in filters:
+                raise BadRequest('Unknown filter: %s=%s' % (key, value))
+            if key.startswith('at'):
+                try:
+                    value = cast(int, value.timestamp())
+                except AttributeError:
+                    raise BadRequest('Time frame filter must be a datetime object: %s=%s' % (key, value))
+            elif key.startswith('bytes') or key == 'lan':
+                value = cast(int, value)
+            elif key == 'accountID':
+                if value == self.myPlexAccount().id:
+                    value = 1  # The admin account is accountID=1
+            params[key] = value
+
+        key = '/statistics/bandwidth?%s' % urlencode(params)
+        data = self.query(key)
+        return self.findItems(data, StatisticsBandwidth)
```

Comment:
Is query and findItems required where for some reason or would it be possible to use fetchItems?